pkg/virt-operator/resource/generate/components/webhooks.go

```diff
@@ -96,6 +96,7 @@ func NewOpertorValidatingWebhookConfiguration(operatorNamespace string) *v1beta1
 				Rules: []v1beta1.RuleWithOperations{{
 					Operations: []v1beta1.OperationType{
 						v1beta1.Update,
+						v1beta1.Create,
```

Comment:
How this one sneaked in? Can you explain how is this related to this change?

Reply:
Before the KubeVirt CR only needed to be validated on update since we where only validating workload placement policy and on creation any workload placement policy is valid. But the validation for `componentsCustomization` could be incorrect on create or update and needs to be check for both events.

Reply:
does "Create" help us here? The operator's validation webhook appears to be created after the KUbeVirt cr is created.

so, it seems the logic looks like. 
- post kubevirt cr
- install kubevirt (and operator validation webhook)

Validation of the kubevirt CR at creation time doesn't seem like it will ever happen due to us never actually having the webhook registered at the point of creation. 

does this sound accurate, or am i missing something

Reply:
You are correct that the CR can be added and processed before the validation webhook is running... Hmm.. is there a better way to do this? My concern is the inconsistency that could happen in the following workflow. 

- User create KubeVirt CR with an invalid patch, it is never applied since it's invalid but the user doesn't know this.
- Five months later
- User goes to update KubeVirt CR setting to add featureflag or some other thing and now they get an error on update about a patch that has been there the whole time. 

Alternatively to this webhook validation I could add `// +kubebuilder:validation:MinLength=1` to the `ResourceType` and `ResourceName` fields to validate nonempty and then could actually get rid of this part of the validation hook completely. However valdiating the patch to be valid JSON i don't think can be done... we do log an error for this maybe that's enough. Do you think this inconsistency between update and create is okay? Or should just the logging be enough for this? or is there another solution I am not thinking of?

Reply:
> User create KubeVirt CR with an invalid patch, it is never applied since it's invalid but the user doesn't know this.

If a broken json patch at creation time prevents the kubevirt installation from occurring, then it can't be ignored.

> Do you think this inconsistency between update and create is okay?

as long as the malformed json patch prevents the install and is reported, then I think this inconsistency will be okay.

So, at creation time, I'd expect a malformed json patch to result in the KubeVirt CR failing to be applied, which will give us a nice human readable string in some condition indicating that a patch is malformed.

And, at update time (assuming the validation webhook and virt-api are online)  we'll catch these issues immediately and not let a malformed json patch get persisted on the KubeVirt CR. 

It's not great, but I'm not seeing a more practical solution right now

Reply:
yeah.. unfortunate but sounds reasonable to me. I have update the PR for reflect these changes.